src/components/application_manager/src/app_launch/app_launch_data_json.cc

```diff
@@ -111,12 +114,22 @@ Json::Value& AppLaunchDataJson::GetApplicationListAndIndex(
   return apps_list;
 }
 
+Json::Value& AppLaunchDataJson::GetApplicationListAndIndex(
+    const ApplicationData& app_data, int32_t& founded_index) const {
+  static Json::Value value;
+  return value;
+}
+
 bool AppLaunchDataJson::IsAppDataAlreadyExisted(
     const ApplicationData& app_data) const {
   LOG4CXX_AUTO_TRACE(logger_);
 
   int32_t index = NotFound;
-  GetApplicationListAndIndex(app_data, index);
+
+  resumption::LastStateAccessor accessor = last_state_wrapper_->get_accessor();
+  Json::Value dictionary = accessor.GetData().dictionary();
```

Comment:
@LitvinenkoIra Don't we need thread safe here?

Reply:
@ZhdanovP take a look at last_state_wrapper implementation.
It's already thread safe